alphas/alpha_functions.py

```diff
@@ -29,3 +31,198 @@ def alpha_001(data: pd.DataFrame, ticker: str) -> pd.DataFrame:
     drop_signal_indices = data["actively_traded"].where(data["actively_traded"] == False).dropna().index
     raw_signal.loc[drop_signal_indices] = 0
     return raw_signal
+
+
+def alpha__011(data: pd.DataFrame, ticker=str):
+    """
+    div{rank[minus(mvwap(5), close)],rank(plus{mvwap(5), close})}
+
+    Data: adjusted close column of given dataframe
+
+    rolling window on mwvwap: 5 days
+    """
+
+    def MVWAP(data: pd.DataFrame, ticker=str):
+        data["Price"] = (data["High"] + data["Low"] + data["Close"]) / 3
+        data["Cumulative_Price_Volume"] = (data["Price"] * data["Volume"]).cumsum()
+        data["Cumulative_Volume"] = data["Volume"].cumsum()
+        data["VWAP"] = data["Cumulative_Price_Volume"] / data["Cumulative_Volume"]
+        data["MVWAP"] = data["VWAP"].rolling(5).mean()
+        data = data.dropna()
+
+        return data
+
+    def rank(x):
+
+        if isinstance(x, np.ndarray):
+            x = pd.Series(x)
+        elif not isinstance(x, (pd.Series, int, float)):
+            raise TypeError(type(x))
+
+        return np.argsort(x) + 1
+
+    data = MVWAP(data)
+    data["Adj Close"] = data["Adj Close"].astype(float)
+    data["MVWAP"] = data["MVWAP"].astype(float)
+    data["Sum"] = data["MVWAP"] + data["Adj Close"]
+    data["Difference"] = data["MVWAP"] - data["Adj Close"]
+    rank_difference = rank(data["Difference"])
+    rank_sum = rank(data["Sum"])
+    raw_signal = rank_difference.div(rank_sum)
+
+    return raw_signal
+
+
+def alpha_012(data: pd.DataFrame, ticker=str):
+    """
+    skew - inv{expmn[bscore(54), 15], 72}
+
+    data: adjusted close column
+    """
+
+    def bscore(data: pd.Dataframe, b: int):
+
+        mean = data["Adj Close"].iloc[-b:].mean()
+        std = data["Adj Close"].iloc[-b:].std()
+        data = data.dropna()
+        bollinger_score = ((data["Adj Close"][-b:] - mean) / std).dropna()
+        return bollinger_score
+
+    def expmn(data: pd.Dataframe, b: int):
+        EWM = ((data.ewm(span=b).mean())).dropna()
```

Comment:
you don't need to initialize a variable here, since it's simple enough to be returned